internal/provider/service_account_tokens_resource.go

```diff
@@ -0,0 +1,222 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	unleash "github.com/Unleash/unleash-server-api-go/client"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var (
+	_ resource.Resource                = &serviceAccountTokensResource{}
+	_ resource.ResourceWithConfigure   = &serviceAccountTokensResource{}
+	_ resource.ResourceWithImportState = &serviceAccountTokensResource{}
+)
+
+func NewServiceAccountTokensResource() resource.Resource {
+	return &serviceAccountTokensResource{}
+}
+
+type serviceAccountTokensResource struct {
+	client *unleash.APIClient
+}
+
+type serviceAccountTokensResourceModel struct {
+	Id               types.Int64  `tfsdk:"id"`
+	ServiceAccountId types.Int64  `tfsdk:"service_account_id"`
+	Secret           types.String `tfsdk:"secret"`
+	Description      types.String `tfsdk:"description"`
+	ExpiresAt        types.String `tfsdk:"expires_at"`
```

Comment:
Notably I've excluded a few fields that I don't think make sense:

- Name: seems to do nothing that description doesn't already do, doesn't seem to have any impact on the token except being displayed adjacent to the token on the UI
- LastSeen: doesn't seem to be useful for a terraform integration
- UserId: also doesn't seem to be useful for a terraform integration

This is really just to reduce public dependencies. These can be added later if they're needed but adding them now means making them sticky. Name in particular seems like it should get dropped on the Unleash side